pkg/apis/core/types.go

```diff
@@ -1966,6 +1966,33 @@ const (
 	PullIfNotPresent PullPolicy = "IfNotPresent"
 )
 
+// ContainerResizePolicy specifies user guidance on how container resource resize should be handled.
+// Only one of the following container resize policies may be specified.
+// If none of the following policies is specified, it defaults to NoRestart.
+type ContainerResizePolicy string
+
+// These are the valid container resize policies:
+// NoRestart policy tells Kubelet to call UpdateContainerResources CRI API to resize
```

Comment:
This needs to be clearer and comments should be grouped with the constants.  E.g. here's my interpretation:

```
// the resources without restarting the container, if possible. This is the default behavior.
// RestartContainer policy tells Kubelet to stop and start the container with when
// new resources are applied. This is needed for legacy applications e.g. java apps
// using -xmxN flag which are unable to use the resized memory without restarting.
const (
	// NoRestart indicates that this container does not require a restart
	// to have a specific resource (e.g. memory) resized.
	NoRestart ContainerResizePolicy = "NoRestart"
	// Restart indicates that this container requires a restart when a
	// specific resource (e.g. memory) is resized.  For example, Java apps
	// which use the `-xmxN` flag would require to be restarted when the
	// memory resource is resized.
	Restart ContainerResizePolicy = "Restart"
)
```


Note I renamed "RestartContainer" since "container" is redundant...

Reply:
Do we need things like RestartOnGrow and RestartOnShrink ?  Should this be RestartAlways?

Reply:
These could be useful, especially RestartOnShrink since reclaiming memory may not be easily accomplished. @derekwaynecarr Should we add these now? I hope it is OK to updated a merged KEP after-the-fact.

Reply:
We can always add them later, as long as we leave room for them.  I don't like designing to hypotheticals, so maybe best to wait until we know we need them?